### ISSUE TITLE
[Procedure Request] Report 698 - Get Sales Orders - InsertReqWkshLineFromSalesLine

### ISSUE THREAD

**Issue by passion4dev** [microsoft/ALAppExtensions#11731]
Dear ALAppExtensions team,

Through this issue we would like to request a **new procedure** `InsertReqWkshLineFromSalesLine`to be added to Report 698 Get Sales Orders with the following signature:
```al
    procedure InsertReqWkshLineFromSalesLine(var ReqLine: Record "Requisition Line"; SalesLine: Record "Sales Line")
    begin
        ReqWkshName."Worksheet Template Name" := vRecReqLine."Worksheet Template Name";
        ReqWkshName.Name := vRecReqLine."Journal Batch Name";

        vRecReqLine.Reset();
        vRecReqLine.SetRange("Worksheet Template Name", ReqWkshName."Worksheet Template Name");
        vRecReqLine.SetRange("Journal Batch Name", ReqWkshName.Name);
        LineNo := 0;
        if vRecReqLine.FindLast() then
            LineNo := vRecReqLine."Line No.";
        SpecOrder := 0;
        If pRecSalesLine."Special Order" then
            SpecOrder := 1;

        InsertReqWkshLine(pRecSalesLine);
        vRecReqLine := ReqLine;
    end;
```
Existing local procedure InsertReqWkshLine(SalesLine: Record "Sales Line") is the hart of the creation of Requistion Lines from Sales Lines with a Purchasing Code in Business Central  (Drop Shipment or Special Order) .

Therefore several Event Publishers are implemented already in this function, and therefore this function **must** be used to create Requistion Lines from Sales Lines with a Purchasing Code.

Currently this function isn't available without starting the report. Publishing this function by a new wrapper function would be a great step forward in terms of expandability.

**Comment by AlexanderYakunin** [comment]
Please clarify what is vRecReqLine, we don't have such global variable. We have ReqLine as global, but you also use it. Can you please adjust you suggestion code to match our code?

**Comment by passion4dev** [comment]
Dear Alexander,

Apologies for this mistake.
The correct new procedure InsertReqWkshLineFromSalesLineto we would like to be added to Report 698 Get Sales Orders:

```al
    procedure InsertReqWkshLineFromSalesLine(var RequisitionLine: Record "Requisition Line"; SalesLine: Record "Sales Line")
    begin
        ReqWkshName."Worksheet Template Name" := RequisitionLine."Worksheet Template Name";
        ReqWkshName.Name := RequisitionLine."Journal Batch Name";

        RequisitionLine.Reset();
        RequisitionLine.SetRange("Worksheet Template Name", ReqWkshName."Worksheet Template Name");
        RequisitionLine.SetRange("Journal Batch Name", ReqWkshName.Name);
        LineNo := 0;
        if RequisitionLine.FindLast() then
            LineNo := RequisitionLine."Line No.";
        SpecOrder := 0;
        If pRecSalesLine."Special Order" then
            SpecOrder := 1;

        InsertReqWkshLine(SalesLine);
        RequisitionLine := ReqLine;
    end;
```

**Comment by AlexanderYakunin** [comment]
Fixed by implementing #12943
